code/clue.js

```diff
@@ -3,57 +3,274 @@
+  const killerAge = document.getElementById('killerAge');
+  const killerOccupation = document.getElementById('killerOccupation');
+  const killerDescription = document.getElementById('killerDescription');
+  const killerFavouriteWeapon = document.getElementById(
+    'killerFavouriteWeapon'
+  );
+
+  mystery.killer = randomSelector(suspects);
+  killerLoader.style.display = 'none';
+  killerHeadLine.innerHTML = 'Killer-information:';
+  killerCard.style.background = mystery.killer.color;
+  killerName.innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  killerImage.src = mystery.killer.image;
+  killerAge.innerHTML = `${mystery.killer.age} years old`;
+  killerOccupation.innerHTML = `${mystery.killer.occupation}`;
+  killerDescription.innerHTML = `${mystery.killer.description}`;
+  killerFavouriteWeapon.innerHTML = `Favourite weapon: ${
+    randomFavouriteWeapon().name
+  }`;
+};
+//function to show weapon information
+const weaponInformation = () => {
+  const weaponName = document.getElementById('weaponName');
+  const weaponWeight = document.getElementById('weaponWeight');
+
+  mystery.weapon = randomSelector(weapons);
+  weaponLoader.style.display = 'none';
+  weaponHeadLine.innerHTML = 'Weapon-information';
+  weaponName.innerHTML = `Weapon: ${mystery.weapon.name}`;
+  weaponWeight.innerHTML = `Weapon weight: ${mystery.weapon.weight}`;
+};
+//function to show the weapon information
+const roomInformation = () => {
+  const roomName = document.getElementById('roomName');
+
+  mystery.room = randomSelector(rooms);
+  roomLoader.style.display = 'none';
+  roomHeadLine.innerHTML = 'Room:';
+  roomName.innerHTML = `${mystery.room}`;
+};
 
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
-
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+  showKillerLoader();
+  setTimeout(killerInformation, 4000);
+};
 
 // CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+const pickWeapon = () => {
+  showWeaponLoader();
+  setTimeout(weaponInformation, 4000);
+};
+
+const pickRoom = () => {
+  showRoomLoader();
+  setTimeout(roomInformation, 4000);
+};
+
+document.getElementById('killerCard').onclick = pickKiller;
+document.getElementById('weaponCard').onclick = pickWeapon;
+document.getElementById('roomCard').onclick = pickRoom;
+
+//The first that shows on the cards
+killerHeadLine.innerHTML = 'Pick a killer';
+weaponHeadLine.innerHTML = 'Pick a weapon';
+roomHeadLine.innerHTML = 'Pick a room';
```

Comment:
For logical order It would have made more sense if these lines came in the beginning of the js-file.

Reply:
I can agree with this 👍